### PR TITLE
ZIOS-11310: No error message when opening a deep link with broken part

### PR DESCRIPTION
**Pull request by nicorsm**
### Issues

App wasn't displaying any error when opening a broken deep link, like `wire://userx/xxx` or `wire://connect/something`.

### Solutions

I've added support to a new deep link error case, `.malformedLink`.

### CODE REVIEW

**Comment by alexaubry** [comment]
The tests seem to be failing still.

File: Source/SessionManager/SessionManagerURLHandler.swift [hunk 1]
```diff
@@ -96,7 +96,8 @@ extension URLAction {
     init?(url: URL, validatingIn defaults: UserDefaults = .shared()) {
         guard let components = URLComponents(string: url.absoluteString),
             let host = components.host,
-            components.scheme == "wire" else {
+            let scheme = components.scheme,
+            scheme.starts(with: "wire") == true else {
```
Comment (alexaubry) [inline Source/SessionManager/SessionManagerURLHandler.swift]: Why do we need this? Our app will never be opened by a URL scheme that is not registered in the Info.plist.

Comment (nicorsm) [reply Source/SessionManager/SessionManagerURLHandler.swift]: There's a test that is trying to call this initializer using a link from google.com. With the last default case that returns an error, the test fails. But yes, both control and test are not used in real life.

**Comment by nicorsm** [comment]
@zenkins release patch

**Pull request merged**